label_studio/tests/test_core.py

```diff
@@ -134,3 +138,51 @@ def test_start_browser():
 
     assert start_browser('http://localhost:8080', True) is None
     assert start_browser('http://localhost:8080', False) is None
+
+@pytest.mark.parametrize('url, block_local_urls, raises_exc', [
+    ('http://0.0.0.0', True, InvalidUploadUrlError),
+    ('http://0.0.0.0', False, None),
+    ('https://0.0.0.0', True, InvalidUploadUrlError),
+    ('https://0.0.0.0', False, None),
+    # Non-http[s] schemes
+    ('ftp://example.org', True, InvalidUploadUrlError),
+    ('ftp://example.org', False, InvalidUploadUrlError),
+    ('FILE:///etc/passwd', True, InvalidUploadUrlError),
+    ('file:///etc/passwd', False, InvalidUploadUrlError),
+    # Start and end of 127.0.0.0/8
+    ('https://127.0.0.0', True, InvalidUploadUrlError),
+    ('https://127.255.255.255', True, InvalidUploadUrlError),
+    # Start and end of 10.0.0.0/8
+    ('http://10.0.0.0', True, InvalidUploadUrlError),
+    ('https://10.255.255.255', True, InvalidUploadUrlError),
+    # Start and end of 172.16.0.0/12
+    ('https://172.16.0.0', True, InvalidUploadUrlError),
+    ('https://172.31.255.255', True, InvalidUploadUrlError),
+    # Start and end of 192.168.0.0/16
+    ('https://192.168.0.0', True, InvalidUploadUrlError),
+    ('https://192.168.255.255', True, InvalidUploadUrlError),
+    # Valid external IPs
+    ('https://4.4.4.4', True, None),
+    ('https://8.8.8.8', True, None),
+    ('http://8.8.8.8', False, None),
+    # Valid external websites
+    ('https://example.org', True, None),
```

Comment:
Flagging for reviewers that the way this test currently works, these hosts will actually be resolved to a real IP address in test. I think an optimal solution would be mocking the implementation of `socket.gethostbyname` in this test, but I could also see an argument for leaving as is.

Reply:
Going to merge as is to avoid further expanding the scope of this diff, but I'll keep an eye on whether we get flakes due to (for example) DNS issues.